publish-mqtt/src/main.rs

```diff
@@ -208,47 +221,67 @@ async fn requests(mut homie: HomieDevice) -> Result<(), Box<dyn Error>> {
             .incoming(INCOMING_TIMEOUT_MS)
             .map(BluetoothEvent::from)
         {
-            let (object_path, value) = match event {
-                Some(BluetoothEvent::Value { object_path, value }) => (object_path, value),
+            match event {
+                Some(BluetoothEvent::Value { object_path, value }) => {
+                    // TODO: Make this less hacky.
+                    if !object_path.ends_with(SERVICE_CHARACTERISTIC_PATH) {
+                        continue;
+                    }
+                    let device_path =
+                        &object_path[..object_path.len() - SERVICE_CHARACTERISTIC_PATH.len()];
```

Comment:
you probably want `object_path.strip_suffix(SERVICE_CHARACTERISTIC_PATH)`
```suggestion
                    let device_path = match object_path.strip_suffix(SERVICE_CHARACTERISTIC_PATH) {
                        Some(path) => path,
                        None => continue,
                    };
```